src/Connections/FeedConnection.ts

```diff
@@ -37,14 +53,19 @@ export class FeedConnection extends BaseConnection implements IConnection {
     static async validateUrl(url: string): Promise<void> {
         try {
             new URL(url);
-            const res = await axios.head(url).catch(_ => axios.get(url));
-            const contentType = res.headers['content-type'];
-            // we're deliberately liberal here, since different things pop up in the wild
-            if (!contentType.match(/xml/)) {
-                throw new Error(`${contentType} doesn't look like an RSS/Atom feed`);
-            }
-        } catch (err) {
-            throw new Error(`${url} doesn't look like a valid feed URL: ${err}`);
+        } catch (ex) {
+            throw new ApiError("Feed URL doesn't appear valid", ErrCode.BadValue);
+        }
+        let res;
+        try {
+            res = await axios.head(url).catch(() => axios.get(url));
+        } catch (ex) {
+            throw new ApiError(`Could not read from URL: ${ex.message}`, ErrCode.BadValue);
+        }
+        const contentType = res.headers['content-type'];
+        // we're deliberately liberal here, since different things pop up in the wild
+        if (!contentType.match(/xml/)) {
+            throw new ApiError(`Feed responded with a content type of "${contentType}", which doesn't look like an RSS/Atom feed`);
```

Comment:
Could include something like 415 (Unsupported media type) in the api error.

Reply:
Done in [6cd3840](https://github.com/matrix-org/matrix-hookshot/pull/459/commits/6cd3840c6104b737c87497f2cca385bfa8456612)